src/offchainapi/protocol.py

```diff
@@ -337,11 +296,52 @@ def apply_response(self, request):
         self.processor.process_command(
             other_addr=other_addr,
             command=request.command,
-            seq=request.cid,
+            cid=request.cid,
             status_success=request.is_success(),
             error=response.error if response.error else None
         )
 
+    def get_dep_locks(self, request):
+        """
+        Get a request object's dependencies lock status, aka reads
+        Args:
+            request: CommandRequestObject, the concerned request
+        Returns:
+            DepLocks: a struct holding missing dependencies, used dependencies
+                and locked dependencies of the concerned request
+        """
+        depends_on_version = request.command.get_dependencies()
+
+        dep_locks = {dv: self.object_locks.try_get(str(dv)) for dv in depends_on_version}
+
+        missing_deps = []
+        used_deps = []
+        locked_deps = []
+        for dep, lock in dep_locks.items():
+            if lock is None:
+                missing_deps.append(str(dep))
+            elif lock == LOCK_EXPIRED:
+                used_deps.append(str(dep))
+            elif lock != LOCK_AVAILABLE:
+                locked_deps.append(str(dep))
```

Comment:
There is a final posibility, namely that the lock has the cid of a command, per the line:
self.object_locks[str(dv)] = request.cid

In that case the lock is available to the command with that cid, and not available to other commands. Is that captured here? This is to make the lock re-entrant: do we need this in this context?

Reply:
good point. Looking at where this is called:
1. sequence_command_local (send a request)
2. handle_request (receive a request)

#1 and #2 are independent and the same cid shouldn't invoke both of the calls (if they really do, that looks like a bug!)  Hence IMO this shouldn't be reentrant. 

What do you think?